addons/l10n_id_efaktur_coretax/tests/test_l10n_id_efaktur_coretax.py

```diff
@@ -0,0 +1,465 @@
+            'invoice_line_ids': [
+                (0, 0, {'name': 'line1', 'price_unit': 100000, 'quantity': 1})
+            ],
+            'l10n_id_kode_transaksi': '08'
+        })
+        out_invoice_2.action_post()
+
+        with self.assertRaises(ValidationError) as val:
+            out_invoice_2.download_efaktur()
+        self._verify_error_message(val, 1, "(Kode 08)")
+
+    # ============================
+    # Test eFaktur XML content
+    # ============================
+
+    def test_efaktur_xml_partner_edit(self):
+        """ Test the effect of changing customer information/fields towards the generated XML content"""
+        self.partner_id.write({
+            "vat": "1234567890999999",
+            "l10n_id_tku": "222222",
+            "l10n_id_buyer_document_type": "Passport",
+            "l10n_id_buyer_document_number": "A123456",
+            "l10n_id_pkp": True
+        })
+
+        out_invoice = self.env["account.move"].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_id.id,
+            'invoice_date': '2019-05-01',
+            'date': '2019-05-01',
+            'invoice_line_ids': [
+                (0, 0, {'product_id': self.product_id.id, 'name': 'line1', 'price_unit': 100000, 'quantity': 1})
+            ],
+            'l10n_id_kode_transaksi': '04'
+
+        })
+        out_invoice.action_post()
+        out_invoice.download_efaktur()
+
+        result_tree = etree.fromstring(out_invoice.l10n_id_coretax_document._generate_efaktur_invoice())
+        expected_tree = self.with_applied_xpath(
+            etree.fromstring(self.sample_xml),
+            '''
+            <xpath expr="//BuyerTin" position="replace">
+                <BuyerTin>0000000000000000</BuyerTin>
+            </xpath>
+            <xpath expr="//BuyerDocument" position="replace">
+                <BuyerDocument>Passport</BuyerDocument>
+            </xpath>
+            <xpath expr="//BuyerDocumentNumber" position="replace">
+                <BuyerDocumentNumber>A123456</BuyerDocumentNumber>
+            </xpath>
+            <xpath expr="//BuyerIDTKU" position="replace">
+                <BuyerIDTKU>1234567890999999222222</BuyerIDTKU>
+            </xpath>
+            '''
+        )
+
+        self.assertXmlTreeEqual(result_tree, expected_tree)
+
+    def test_efaktur_xml_trx_04(self):
+        """ Test result of regular valid invoice with 04 transaction code.
+        Expected to see OtherTaxBase having 11/12 of the actual TaxBase calculated
+        """
+
+        out_invoice = self.env["account.move"].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_id.id,
+            'invoice_date': '2019-05-01',
+            'date': '2019-05-01',
+            'invoice_line_ids': [
+                (0, 0, {'product_id': self.product_id.id, 'name': 'line1', 'price_unit': 100000, 'quantity': 1})
+            ],
+            'l10n_id_kode_transaksi': '04'
+
```

Comment:
```suggestion
```
I won't tag each of these but try to avoid empty lines that are there for nothing